cpp/oneapi/dal/backend/primitives/sort/sort_dpc.cpp

```diff
@@ -277,273 +48,88 @@ sycl::event radix_sort_indices_inplace<Float, Index>::operator()(ndview<Float, 1
-                                     elem_count_,
-                                     bit_offset,
-                                     local_size_,
-                                     local_hist_count_,
-                                     hist_scan_deps);
-        }
-    }
-
-    sort_event_ = res_deps;
-    return res_deps;
-}
-
-template <typename Integer>
-radix_sort<Integer>::radix_sort(const sycl::queue& queue) : queue_(queue),
-                                                            vector_count_(0) {}
-
-template <typename Integer>
-radix_sort<Integer>::~radix_sort() {
-    sort_event_.wait_and_throw();
-}
-
-template <typename Integer>
-void radix_sort<Integer>::init(sycl::queue& queue, std::int64_t vector_count) {
-    ONEDAL_ASSERT(vector_count > 0);
-    ONEDAL_ASSERT(vector_count <= de::limits<std::uint32_t>::max());
-
-    const std::uint32_t uint_vector_count = de::integral_cast<std::uint32_t>(vector_count);
-    if (vector_count_ != uint_vector_count) {
-        vector_count_ = uint_vector_count;
-
-        buffer_ = ndarray<Integer, 2>::empty(queue_,
-                                             { vector_count_, radix_range_ },
-                                             sycl::usm::alloc::device);
-    }
+    auto event = oneapi::dpl::experimental::kt::gpu::esimd::radix_sort_by_key<true, 8>(
+        queue,
+        val_in.get_mutable_data(),
+        val_in.get_mutable_data() + val_in.get_count(),
+        ind_in.get_mutable_data(),
+        dpl::experimental::kt::kernel_param<256, 32>{});
+    return event;
 }
 
 template <typename Integer>
-sycl::event radix_sort<Integer>::operator()(ndview<Integer, 2>& val_in,
-                                            ndview<Integer, 2>& val_out,
-                                            std::int64_t sorted_elem_count,
-                                            const event_vector& deps) {
-    ONEDAL_PROFILER_TASK(sort.radix_sort, queue_);
-    // radixBuf should be big enough to accumulate radix_range elements
-    ONEDAL_ASSERT(val_in.get_dimension(1) > 0);
-    ONEDAL_ASSERT(sorted_elem_count > 0);
-    ONEDAL_ASSERT(val_in.get_dimension(0) == val_out.get_dimension(0));
-    ONEDAL_ASSERT(val_in.get_dimension(1) == val_out.get_dimension(1));
-    ONEDAL_ASSERT(val_out.has_mutable_data());
-
-    if (val_in.get_dimension(0) > de::limits<std::uint32_t>::max()) {
-        throw domain_error(dal::detail::error_messages::invalid_range_of_rows());
+sycl::event radix_sort(sycl::queue& queue,
+                       ndview<Integer, 2>& val_in,
+                       ndview<Integer, 2>& val_out,
+                       std::int64_t sorted_elem_count,
+                       const event_vector& deps) {
+    ONEDAL_PROFILER_TASK(sort.radix_sort, queue);
+
+    const auto row_count = val_in.get_dimension(0);
+    const auto col_count = val_in.get_dimension(1);
+    sycl::event radix_sort_event;
+
+    for (std::int64_t row = 0; row < row_count; ++row) {
```

Comment:
Looks like the previous implementation can be more performant for the matrices with large number of rows and small number of columns.

I would not delete it, but provide some performance considerations when it is preferred to use the old implementation, and when - the new one.